app/routes/vehicle.py

```diff
@@ -0,0 +1,85 @@
+from fastapi import APIRouter, HTTPException, status
+from sqlalchemy.exc import IntegrityError
+from datetime import datetime
+from typing import List
+
+from ..models.schemas import VehicleCreate, VehicleUpdate, VehicleCreateOut, VehicleOut
+from ..models.models import Vehicle
+from ..dependencies.db_connection import DatabaseDependency
+from ..dependencies.oauth2 import CurrentActiveUserDependency
+
+import base64
+
+router = APIRouter(
+    prefix='/api/vehicles',
+    tags=['Vehicles']
+)
+
+@router.get('/', response_model=List[VehicleOut], status_code=status.HTTP_200_OK)
+def get_all_vehicles(current_user: CurrentActiveUserDependency, db: DatabaseDependency):
+    query = db.query(Vehicle).filter(Vehicle.is_deleted == False)
+    if not current_user.is_superuser:
```

Comment:
check quyền trước để không tốn thời gian query db

Reply:
@router.get('/', response_model=List[VehicleOut], status_code=status.HTTP_200_OK)
def get_all_vehicles(current_user: CurrentActiveUserDependency, db: DatabaseDependency):
    query = db.query(Vehicle).filter(Vehicle.is_deleted == False)
    if not current_user.is_superuser:
        query = query.filter(Vehicle.owner_id == current_user.id)
    vehicles = query.all()
    return vehicles
    
    
Cái này không lo, nó lâu hay chậm là khi query thành ToList(), nên cứ thiết lập query cho ổn, là ToList() nó ra trong lần cuối là được

Reply:
với admin hiển thị cả deleted nhé

Reply:
Hiển thị delete chỉ đối với user thôi, mấy cái bản ghi của các bảng khác không cần thiết để cho admin thấy đâu